kasa/module.py

```diff
@@ -132,6 +137,44 @@ def __init__(self, device: Device, module: str):
         self._device = device
         self._module = module
         self._module_features: dict[str, Feature] = {}
+        self._bound_feature_ids: dict[property, str] = {}
+
+    def _bound_feature_id(self, attr: property) -> str | None:
+        """Get bound feature for module or none if not supported."""
+        if attr in self._bound_feature_ids:
+            return self._bound_feature_ids[attr]
+        if isinstance(attr, property):
+            hints = get_type_hints(attr.fget, include_extras=True)
+            if (return_hints := hints.get("return")) and hasattr(
+                return_hints, "__metadata__"
+            ):
+                metadata = hints["return"].__metadata__
+                for meta in metadata:
+                    if isinstance(meta, FeatureProperty):
+                        self._bound_feature_ids[attr] = meta.id
+                        return meta.id
+        return None
+
+    def is_bound_feature(self, attr: property) -> bool:
+        """Return True if property is bound to a feature."""
+        return bool(self._bound_feature_id(attr))
+
+    def has_bound_feature(self, attr: property) -> bool:
+        """Return True if the bound property feature is supported."""
+        if id := self._bound_feature_id(attr):
+            return id in self._module_features
+        raise KasaException("")
+
+    def get_bound_feature(self, attr: property) -> Feature | None:
+        """Get Feature for a bound property or None if not supported."""
+        if id := self._bound_feature_id(attr):
+            return self._module_features.get(id)
+        raise KasaException("")
+
+    @property
+    def module(self) -> type[Self]:
```

Comment:
```suggestion
    def type(self) -> type[Self]:
```
Or `module_type`?

Reply:
How about `module_class`? If not then I prefer `module_type`

Reply:
Or do we really need this in the end? Accessing __class__ directly should be fine, no?

Reply:
It just makes the api clunky that way.  This looks nicer:
```
fan_speed_level_feature = fan.get_bound_feature(fan.module.fan_speed_level)
```
Than this:
```
fan_speed_level_feature = fan.get_bound_feature(fan.__class__.fan_speed_level)
```
or
```
fan_speed_level_feature = fan.get_bound_feature(type(fan).fan_speed_level)
```

Reply:
Or maybe we should pass the property name as a string instead:
```
fan_speed_level_feature = fan.get_bound_feature("fan_speed_level")
```

Reply:
Dropped in favour of passing a string which is more consistent with builtin `hasattr` and `getattr` pattern.